src/libstore/include/nix/store/globals.hh

```diff
@@ -1237,6 +1237,23 @@ public:
           Set it to 1 to warn on all paths.
         )"
     };
+
+    struct ExternalBuilder
+    {
+        std::vector<std::string> systems;
+        Path program;
```

Comment:
@edolstra What do you think about adding an `arguments` field to this? The semantics could be "the builder JSON document is provided at the last position", so you could use it like `/bin/foo --somearg true --anotherarg false --builder-json <NIX PUTS THE JSON DOC PATH HERE>`

That would make it much more flexible with regards to less-trivial external builders since it would allow specifying arguments directly in this JSON setting rather than needing to create a wrapper script.

Reply:
I opened https://github.com/DeterminateSystems/nix-src/pull/120 for my approach to this